command/util_test.go

```diff
@@ -56,71 +52,55 @@ func testAgentWithAPIClient(t *testing.T) (*agentWrapper, *api.Client) {
 	return agent, client
 }
 
-func testAgentWithConfig(t *testing.T, cb func(c *agent.Config)) *agentWrapper {
-	return testAgentWithConfigReload(t, cb, nil)
-}
-
-func testAgentWithConfigReload(t *testing.T, cb func(c *agent.Config), reloadCh chan chan error) *agentWrapper {
-	lw := logger.NewLogWriter(512)
+func testAgentWithConfig(t *testing.T, cb func(c *agent.Config)) *server {
 	conf := nextConfig()
 	if cb != nil {
 		cb(conf)
 	}
 
-	dir := testutil.TempDir(t, "agent")
-	conf.DataDir = dir
-
-	a, err := agent.Create(conf, lw, nil, reloadCh)
+	conf.DataDir = testutil.TempDir(t, "agent")
+	a, err := agent.NewAgent(conf)
 	if err != nil {
-		os.RemoveAll(dir)
-		t.Fatalf(fmt.Sprintf("err: %v", err))
+		os.RemoveAll(conf.DataDir)
+		t.Fatal("Error creating agent:", err)
 	}
-
-	conf.Addresses.HTTP = "127.0.0.1"
-	httpAddr := fmt.Sprintf("127.0.0.1:%d", conf.Ports.HTTP)
-	http, err := agent.NewHTTPServers(a, conf, os.Stderr)
-	if err != nil {
-		os.RemoveAll(dir)
-		t.Fatalf(fmt.Sprintf("err: %v", err))
+	a.LogOutput = logger.NewLogWriter(512)
+	if err := a.Start(); err != nil {
+		os.RemoveAll(conf.DataDir)
+		t.Fatalf("Error starting agent: %v", err)
 	}
 
-	if http == nil || len(http) == 0 {
-		os.RemoveAll(dir)
-		t.Fatalf(fmt.Sprintf("Could not create HTTP server to listen on: %s", httpAddr))
-	}
-
-	return &agentWrapper{
-		dir:      dir,
-		config:   conf,
-		agent:    a,
-		http:     http[0],
-		httpAddr: httpAddr,
-	}
+	conf.Addresses.HTTP = "127.0.0.1"
+	addr := fmt.Sprintf("%s:%d", conf.Addresses.HTTP, conf.Ports.HTTP)
+	return &server{agent: a, config: conf, httpAddr: addr, dir: conf.DataDir}
 }
 
-func nextConfig() *agent.Config {
-	idx := int(atomic.AddUint64(&offset, 1))
-	conf := agent.DefaultConfig()
+var nextPort uint64 = 10000
 
+func nextConfig() *agent.Config {
```

Comment:
I'm working on that right now. That and then the code in `command/server_test.go`